README.md

```diff
@@ -14,3 +14,19 @@ part of the array each recursive call considers.
 Analyse the time complexity of your implementation and give a $\Theta$ bound for
 its worst-case runtime. Add your answer, including your reasoning, to this
 markdown file.
+
+$\Theta$ bound for my implementation (and other mergesort implementations) is $\Theta$(nlog(n))
+For my implementation this is because the outer loop works over arrays that double in size each loop
+halfing the amount of time taken each loop, leading to a time complexity of log(n)
+The inner loop takes linear time as it iterates over all elements of the array each pass
+to compare and sort during the merge process.
+Thus we do $\Theta$(n) work $\Theta$(log(n)) times, making the resulting time complexity:
+$\Theta$(nlog(n))
+
+I used the lecture material, as well as a previous mergesort I had worked with as a base
+for my code and had chatGPT's help adjusting it to work in place.
```

Comment:
What specifically did ChatGPT do?

Reply:
It helped with adjusting my algorithm from recursive to iterative because functional has my brain so stuck up on recursion that doing things iteratively again has been a whole ordeal. I also typically do a bit of back and forth with chatGPT once I have a fully written function to work out the kinks (usually fails on edge cases).

Reply:
So what part of the code you've submitted would you say was written by ChatGPT?

Reply:
ChatGPT helped write the code within the original mergesort function, specifically the idea of breaking our problem down to subarrays of size one and doubling at each step instead of the typical recursion of breaking the problem in half at each step. It also helped with some simple syntax stuff such as shifting the values in the array to make it in place. Having now completed almost all of the required programs for this class this seems like very trivial things to have needed aid for and I'm confident I could now complete this assignment with much less help, but wanted to be fully transparent about the help I received at the time of completion.